src/rpc/mining.cpp

```diff
@@ -906,16 +911,36 @@ UniValue getblocksubsidy(const UniValue& params, bool fHelp)
 
     UniValue result(UniValue::VOBJ);
     if (canopyActive) {
+        KeyIO keyIO(Params());
         UniValue fundingstreams(UniValue::VARR);
         auto fsinfos = Consensus::GetActiveFundingStreams(nHeight, consensus);
-        for (auto fsinfo : fsinfos) {
+        for (int idx = 0; idx < fsinfos.size(); idx++) {
+            auto fsinfo = fsinfos[idx];
             CAmount nStreamAmount = fsinfo.Value(nBlockSubsidy);
             nMinerReward -= nStreamAmount;
 
             UniValue fsobj(UniValue::VOBJ);
             fsobj.pushKV("recipient", fsinfo.recipient);
             fsobj.pushKV("specification", fsinfo.specification);
             fsobj.pushKV("value", ValueFromAmount(nStreamAmount));
+            fsobj.pushKV("valueZat", nStreamAmount);
+
+            auto fs = consensus.vFundingStreams[idx];
+            auto address = fs.get().RecipientAddress(consensus, nHeight);
+
+            CScript* outpoint = boost::get<CScript>(&address);
+            libzcash::SaplingPaymentAddress* zaddr = boost::get<libzcash::SaplingPaymentAddress>(&address);
+            UniValue pubkey(UniValue::VOBJ);
+
+            if (outpoint != nullptr) {
+                // For transparent funding stream addresses
+                ScriptPubKeyToJSON(*outpoint, pubkey, true);
+            } else if (zaddr != nullptr) {
+                // For shielded funding stream addresses
+                pubkey.pushKV("address", keyIO.EncodePaymentAddress(*zaddr));
```

Comment:
Is this (funding to a shielded address) being tested? I don't see where it is. It may not be possible to test this. Actually, it may not be reasonable for a funding stream ever to go to a shielded address for reasons of transparency and accountability. It may be better to remove this code (just not print anything in this case, where `outpoint` is null), and let this code be updated when and if a funding stream needs to go to a zaddr. 

In general, it's not good to have "dead code" that can't even be tested. Perhaps just add a comment saying the funding streams to shielded addresses isn't yet supported.

Reply:
there are several instances in the codebase where we account for a shielded funding stream address. e.g. https://github.com/zcash/zcash/blob/master/src/main.cpp#L941-L951

```

// ZIP 207: detect shielded funding stream elements
if (canopyActive) {
    libzcash::SaplingPaymentAddress zaddr(encPlaintext->d, outPlaintext->pk_d);
    for (auto it = fundingStreamElements.begin(); it != fundingStreamElements.end(); ++it) {
        const libzcash::SaplingPaymentAddress* streamAddr = boost::get<libzcash::SaplingPaymentAddress>(&(it->first));
        if (streamAddr && zaddr == *streamAddr && encPlaintext->value() == it->second) {
            fundingStreamElements.erase(it);
            break;
        }
    }
}
```

Reply:
ZIP 207 intentionally allows shielded (Sapling) funding stream addresses, so this should be supported.